crates/utils/src/error.rs

```diff
@@ -9,49 +10,192 @@ struct ApiError {
   error: String,
 }
 
+#[derive(EnumString, Display, Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+#[serde(tag = "error_type", content = "message", rename_all = "snake_case")]
```

Comment:
Don't rename any of these to snake_case, we don't do that anywhere else and prefer both the Rust code and other client types to have the same text.

Reply:
I made it PascalCase. I think it could go either way though. If I understood the serde docs correctly, this enum should be serialized as `{ error_type: CaptchaIncorrect, message: undefined }`. So the enum types should be json keys, but json values instead.

Correct me if I don't understand serde correctly.

Reply:
You can make a simple test case which calls `serde_json::to_string` and compares against an expected json string. That way you see what the output is.

Reply:
@dessalines The tests confirm that the errors are serialized in the way I anticipated earlier in this thread. PascalCase and snake_case should both work: up to you and @Nutomic which you prefer.

Reply:
It has to be snake_case, otherwise its a breaking change.